tox.ini

```diff
@@ -138,44 +143,93 @@ changedir = {[base]changedir}
 commands =
     coveralls
 
-[testenv:ensure_non_running_provider]
+# LSR_MOLECULE_DEPS may contain aditional Molecule dependencies. For example,
+# in `network` system role, LSR_MOLECULE_DEPS can be set as
+#
+#   LSR_MOLECULE_DEPS='-rmolecule_requirements.txt'
+#
+# where `molecule_requirements.txt` contains two lines:
+#
+#   jmespath
+#   selinux
+#
+[molecule]
 deps =
-    PyYAML
-changedir = {toxinidir}/tests
-commands = {toxinidir}/tests/ensure_non_running_provider.py
+    docker
+    molecule
+    {env:LSR_MOLECULE_DEPS:}
 
 [testenv:molecule_lint]
+envdir = {toxworkdir}/env-{env:TRAVIS_PYTHON_VERSION:molecule}
+deps =
+    {[molecule]deps}
 commands_pre =
     molecule --version
     ansible --version
-commands = molecule {posargs} lint
+commands =
+    molecule {posargs} lint
 
 [testenv:molecule_syntax]
-commands = molecule {posargs} syntax
+envdir = {toxworkdir}/env-{env:TRAVIS_PYTHON_VERSION:molecule}
+deps =
+    {[molecule]deps}
+commands =
+    molecule {posargs} syntax
 
 [testenv:molecule_test]
-commands = molecule {posargs} test
+envdir = {toxworkdir}/env-{env:TRAVIS_PYTHON_VERSION:molecule}
+deps =
+    {[molecule]deps}
+commands =
+    molecule {posargs} test
+
+# Here we provide a way how a role can add its custom command to be run. Such
+# extra command is run at the end of each testenv run and is driven by
+# environment variables. Involved environment variables are:
+#
+#   LSR_TEXTRA_DEPS
+#     - contains dependency needed by commands to run smoothly; if more than
+#       one dependency is needed, use external file together with '-r' option
+#       (see PEP 508)
+#
+#   LSR_TEXTRA_DIR
+#     - directory to which to cd
+#
+#   LSR_TEXTRA_CMD
+#     - custom command to be run
+#
+# Example: `network` system role need to run `./tests/ensure_non_running_provider.py`
+#          to check for the existence of `*_provider.yml` playbooks. The script
+#          is run in Python 3.6.
+#
+#          To make this possible, we add to `.travis/config.sh` a snippet:
+#
+#              export LSR_TEXTRA_DEPS='PyYAML'
+#              export LSR_TEXTRA_DIR='tests'
+#              export LSR_TEXTRA_CMD='./ensure_non_running_provider.py'
+#
+[testenv:extra]
+envdir = {toxworkdir}/env-{env:TRAVIS_PYTHON_VERSION:extra}
+deps =
+    {env:LSR_TEXTRA_DEPS:}
+passenv = *
+changedir = {toxinidir}/{env:LSR_TEXTRA_DIR:.}
+commands =
+    {env:LSR_TEXTRA_CMD:python --version}
 
 [pytest]
 addopts = -rxs
 
 [flake8]
-show_source = True
+show_source = true
 max-line-length = 88
 ignore = E402,W503
 
-[pylint]
```

Comment:
After removing these options, are they still picked up somehow by IDEs? AFAIK tox.ini is a standard file that is used by many tools to allow to configure the code checkers used with vim-python for example.

Reply:
It depends on IDE imho. I don't know much about vim-python, but if it prefers `tox.ini` over another config files (like `pylintrc`), I will return these options back.